components/server/src/user/user-service.ts

```diff
@@ -262,25 +262,33 @@ export class UserService {
         return user;
     }
 
-    async updateUserOnLogin(user: User, authUser: AuthUser, candidate: Identity, token: Token) {
+    async updateUserOnLogin(user: User, authUser: AuthUser, candidate: Identity, token: Token): Promise<User> {
         // update user
         user.name = user.name || authUser.name || authUser.primaryEmail;
         user.avatarUrl = user.avatarUrl || authUser.avatarUrl;
         await this.onAfterUserLoad(user);
         await this.updateUserIdentity(user, candidate);
         await this.userDb.storeSingleToken(candidate, token);
+
+        const updated = await this.userDb.findUserById(user.id);
+        if (!updated) {
+            throw new Error("User does not exist");
+        }
+        return updated;
     }
 
     async onAfterUserLoad(user: User): Promise<User> {
         return user;
     }
 
     async updateUserIdentity(user: User, candidate: Identity) {
+        log.info("Updating user identity", {
+            user,
+            candidate,
+        });
         // ensure single identity per auth provider instance
         user.identities = user.identities.filter((i) => i.authProviderId !== candidate.authProviderId);
         user.identities.push(candidate);
-
-        await this.userDb.storeUser(user);
```

Comment:
why removing this? doesn't this break `updateUserOnLogin`?

Reply:
Not sure how the change came about. We can re-add

Reply:
re-added